### DISCUSSION TITLE
Regarding mirroring types from other crates

### DISCUSSION THREAD

**Discussion by escwxyz** [fzyzcjy/flutter_rust_bridge#1161] [General]
Hi, 

I am not sure if I understand correctly, I have to mirror all the types from other crates that are returned, right? If it's deeply nested, recursively. How to handle that?

For example:

```rust

#[tokio::main(flavor = "current_thread")]
pub async fn init() -> Result<Arc<Surreal<Client>>> {
    tracing_subscriber::fmt().init();

    info!("Initializing");
    info!("Setting up database...");

    let db = Surreal::new::<Ws>("localhost:8000").await?;

    db.signin(Root {
        username: "root",
        password: "root",
    })
    .await?;

    db.use_ns("development").use_db("dev").await?;

    let client = Arc::new(db);

    info!("Finished initializing");

    Ok(client)
}
```

I have to mirror the `Surreal` and `Client` struct, where `Surreal` is like this:

```rust
pub struct Surreal<C: Connection> {
	router: OnceCell<Arc<Router<C>>>,
}
```
So I need to mirror the `Connection` and `Router` as well, and so on?

**Comment by fzyzcjy** [comment]
cc @Unoqwy who implemented that helpful mirroring feature :)

Btw, it is also great if we can somehow directly handle things from other crates without mirroring. Feel free to PR!
